src/editorKit.ts

```diff
@@ -170,9 +174,22 @@ export default class EditorKitBase {
 
       this.delegate.setEditorRawText(text)
 
+      if (this.delegate.onNoteContentChange) {
+        const previousContent = previousNote?.content
+        const newContent = this.note?.content
+
+        if (previousContent != newContent) {
```

Comment:
Then how about if we just check `content.text`? That is what the editor will be mostly changing.

```suggestion
        if (this.previousText !== text) {
```

Reply:
Eh, I don't know about that, especially because you're calling `onNoteContentChange` only for text change, which is misleading.

Do components need this new delegate function? Or was it just a nice to have? If not needed then we can omit altogether.